server/src/main/java/org/candlepin/dto/api/v1/ConsumerContentOverrideDTO.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2009 - 2018 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+
+
+/**
+ * The ConsumerContentOverrideDTO is used as a DTO for the ConsumerContentOverride
+ * instances.
+ */
+public class ConsumerContentOverrideDTO extends ContentOverrideDTO<ConsumerContentOverrideDTO> {
+    public static final long serialVersionUID = 1L;
+
+    protected ConsumerDTO consumer;
+
+
+    /**
+     * Initializes a new ConsumerContentOverrideDTO instance with null values.
+     */
+    public ConsumerContentOverrideDTO() {
+        // Intentionally left empty
+    }
+
+    /**
+     * Initializes a new ConsumerContentOverrideDTO instance which is a shallow copy of the provided
+     * source entity.
+     *
+     * @param source
+     *  The source entity to copy
+     */
+    public ConsumerContentOverrideDTO(ConsumerContentOverrideDTO source) {
+        this.populate(source);
+    }
+
+    /**
+     * Fetches the activation key that's overriding the content. If the activation key has not been
+     * set, this method returns null.
+     *
+     * @return
+     *  The activation key overriding the content, or null if the activation key has not been set
+     */
+    @JsonIgnore
+    public ConsumerDTO getConsumer() {
+        return this.consumer;
+    }
+
+    /**
+     * Sets or clears the activation key that's overriding the content. If the provided key is null,
+     * any key currently set will be cleared.
+     *
+     * @param consumer
+     *  The activation key that's overriding the content, or null to clear the key
+     *
+     * @return
+     *  A reference to this DTO
+     */
+    @JsonProperty("key")
```

Comment:
Fixed.

Reply:
I did a quick lookup and it looks to me that the tests aren't actually setting this field themselves (at least the couple I checked. I didn't do a thorough check). But then I looked at ContentOverrideResource and it looks like the resources themselves aren't even looking for this field, and instead, they're retrieving the uuid/id from the URL (in the case of consumerCO they retrieve the consumer_uuid, and in the case of the akCO they're using the activation_key_id), and then load the actual entity from the db:
 https://github.com/candlepin/candlepin/blob/7099ceee48270e8d153c5858192f4792f05be68e/server/src/main/java/org/candlepin/resource/ContentOverrideResource.java#L93-L95

Currently running the spec tests locally with this field completely ignored in both DTOs since it seems they aren't needed on client input.